### PR TITLE
Fix call of the burrowed unintentional larva queue leaves

### PR DESCRIPTION
**Pull request by Wayland-Smithy**
## About The Pull Request

Since `attempt_to_join_as_larva()` acts like a toggle now if you are already in the queue when you click the link in chat to Join as a Larva the result will be you get booted as if you clicked the regular button.

This adds a check to prevent that from happening.

Fixes: #10063

## Why It's Good For The Game

Fix good.

## Changelog
:cl:
fix: Fixed call of the burrowed unintentional larva queue leaves.
/:cl:

### CODE REVIEW

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -200,6 +200,11 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 			return
 		var/mob/dead/observer/ghost = usr
 
+		var/datum/hive_status/normal/HS = GLOB.hive_datums[XENO_HIVE_NORMAL]
+		if(LAZYFIND(HS.candidate, ghost))
+			to_chat(usr, span_warning("You are already in the queue to become a Xenomorph."))
+			return
```
Comment (TiviPlus) [inline code/modules/mob/dead/observer/observer.dm:206]:
```suggestion
			to_chat(ghost, span_warning("You are already in the queue to become a Xenomorph."))
			return
```

**Review by TiviPlus — APPROVED** [review]
rest lgtm

**Review by BraveMole — CHANGES_REQUESTED** [review]
Does this verb needs to exist ?

**Comment by Wayland-Smithy** [comment]
Idk, and I don't think that is a good reason to block the PR. It can always be deleted wholesale later.

**Comment by TiviPlus** [comment]
Fix now remove later

**Pull request merged**